source/developers-guide/event-guide/index.md

```diff
@@ -218,109 +320,147 @@ Some real world examples for `filter` are:
+namespace SwagEvents\Subscriber;
 
-Subscribers are basically custom classes which register to Shopware events. This way, the Bootstrap doesn't know about the
-events, which can, in turn, be encapsulated in corresponding classes. Let's have a look:
+use Doctrine\Common\Collections\ArrayCollection;
+use Enlight\Event\SubscriberInterface;
+use SwagEvents\Components\NameClass3;
+use SwagEvents\Components\NameClass4;
 
-```
-class Shopware_Plugins_Frontend_MyPlugin_Bootstrap extends Shopware_Components_Plugin_Bootstrap
+class SwagEventsSubscriber implements SubscriberInterface
 {
-    public function install()
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
     {
-        $this->subscribeEvent('Enlight_Controller_Front_StartDispatch', 'onRegisterSubscriber');
-        $this->subscribeEvent('Shopware_Console_Add_Command', 'onRegisterSubscriber');
+        return [
+            'SwagEvent_Controller_collectAction' => 'onCollect',
+        ];
+    }
 
-        return true;
+    /**
+     * @param \Enlight_Event_EventArgs $args
+     *
+     * @return ArrayCollection
+     */
+    public function onCollect(\Enlight_Event_EventArgs $args)
+    {
+        return new ArrayCollection(
+            [
+                new NameClass3(),
+                new NameClass4()
+            ]
+        );
     }
 }
 ```
 
-In the `install` method of the plugin an early Shopware event called `Enlight_Controller_Front_StartDispatch` is subscribed.
-It is also necessary to always subscribe to the `Shopware_Console_Add_Command`.
-So still these events are needed to bring my subscribers into play - but afterwards, most other events can be
-registered using subscribers.
+The second step is to register the subscriber class in the `PluginBaseDirectory/Resources/service.xml`.
+To say shopware the the registered service is a subscriber::class add the `shopware.event_subscriber` tag.
+```xml<tag name="shopware.event_subscriber"/>```
 
-The event callback `onRegisterSubscriber` usually looks like this:
+```xml
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
+    <services>
+        <service id="swag_events.subscriber.swag_event_subscriber" class="SwagEvents\Subscriber\SwagEventsSubscriber">
+            <tag name="shopware.event_subscriber"/>
+        </service>
+    </services>
+</container>
 ```
-public function onRegisterSubscriber(Enlight_Event_EventArgs $args)
-{
-    Shopware()->Events()->addSubscriber(new \Shopware\MyPlugin\MySubscriber());
-}
 
-```
+(in this case it is `onCollect`). The shopware event manager will now automatically call this callback
+as soon as the event `SwagEvent_Controller_collectAction` occurs.
+
+The services (subscriber) in the service.xml will registered automatically by installing the plugin. 
```

Comment:
in the service**s**.xml will be registered

Reply:
d